docs/core/deploying/preparing-libraries-for-trimming.md

```diff
@@ -0,0 +1,188 @@
+
+Note that the analysis results depend on the implementation details of your dependencies. If you update to a new version of a dependency, this may introduce analysis warnings if the new version added non-understood reflection patterns, even if there were no API changes. In other words, introducing trim analysis warnings to a library is a breaking change when the library is used with `PublishTrimmed`.
+
+## Resolving trim warnings
+
+The above steps will produce warnings about code that may cause problems when used in a trimmed app. Here are a few examples of the most common kinds of warnings you may encounter, with recommendations for fixing them.
+
+### RequiresUnreferencedCode
+
+```csharp
+using System.Diagnostics.CodeAnalysis;
+
+public class MyLibrary
+{
+    public static void Foo()
+    {
+        // warning IL2026 : MyLibrary.Foo: Using method 'MyLibrary.Bar' which has
+        // 'RequiresUnreferencedCodeAttribute' can break functionality
+        // when trimming application code.
+        Bar();
+    }
+
+    [RequiresUnreferencedCode("Bar does something incompatible with trimming.")]
+    static void Bar()
+    {
+    }
+}
+```
+
+This means the library calls a method which has explicitly been annotated as incompatible with trimming, using [`RequiresUnreferencedCodeAttribute`](
+https://docs.microsoft.com/dotnet/api/system.diagnostics.codeanalysis.requiresunreferencedcodeattribute?view=net-5.0). To get rid of the warning, consider whether `Foo` needs to call `Bar` to do its job. If so, annotate the caller `Foo` with `RequiresUnreferencedCode` as well; this will "bubble up" the warning so that callers of `Foo` get a warning instead:
+
+```csharp
+    // Warn for calls to Foo, but not for Foo's call to Bar.
+    [RequiresUnreferencedCode("Calls Bar.")]
+    public static void Foo()
+    {
+        // ...
+    }
+```
+
+Once you have "bubbled up" the attribute all the way to public APIs (so that these warnings are produced only for public methods, if at all), you are done. Apps which call your library will now get warnings if they call those public APIs, but these will no longer produce warnings like `IL2104: Assembly 'MyLibrary' produced trim warnings`.
+
+### DynamicallyAccessedMembers
+
+```csharp
+using System.Diagnostics.CodeAnalysis;
+
+public class MyLibrary
+{
+    public static void Foo(Type type)
+    {
+        // warning IL2067: MyLibrary.Foo(Type): 'type' argument does not satisfy
+        // 'DynamicallyAccessedMemberTypes.PublicMethods' in call to 'MyLibrary.UseMethods(Type)'.
+        // The parameter 't' of method 'MyLibrary.Foo(Type)' does not have matching annotations.
+        UseMethods(type);
+    }
+
+    static void UseMethods(
+        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicMethods)]
```

Comment:
I think it is better to start with a method that straight uses Reflection, teach how to annotate that method with `DynamicallyAccessedMembers`. Then the next step is to show other methods calling that first method, and how to annotate them.